test/common/README.md

```diff
@@ -299,6 +299,14 @@ If `fn` is not provided, an empty function will be used.
 Returns a function that triggers an `AssertionError` if it is invoked. `msg` is
 used as the error message for the `AssertionError`.
 
+### `mustNotMutate(object)`
+
+* `object` [\<Object>][<Object>] | [\<Function>][<Function>]
```

Comment:
What's the reason behind accepting functions? There doesn't seem to be a test that passes a function here. Also, doesn't it feel a little out of place to accept a function in a variable named `object`?

Reply:
That's solely because functions are mutable and potentially can have nested mutable properties. I agree that it looks odd, but not sure how to word it properly. Technically, this function accepts anything, and for primitive values returns them "as is" since they are immutable by design. What is the best way to express that in docs?

I'll add functions in test, thanks for noticing.

Reply:
I don't think this should account for primitive types because as you said, those are immutable already and there's no need to test that.

If we do want to support both functions and objects, I think `target` would be a better name (I stole that from the naming of the first argument of [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)) but I would suggest that for now this should accept just objects and later on we can add support for functions in case the need arises (there isn't one right now because existing tests don't test the immutability of functions passed in as args).

Reply:
Thanks, `target` is perfect! In code this name is used by handlers, so I guess `original` would be acceptable.

Yes, agreed on functions, I'll revert their support until there will be a practical need for it or ability to fully-transparent support.

For future references: `mustNotMutate()` might need configurable options, such as optional recursion limit, disabling particular checks (e.g. defineProperty), `get()` call counter, etc. Adding such options should depend on what is needed in real-life scenarios.
Since the most common meaning for this function is to tell "this is user input, do not touch anything, just read known-for-you properties", one possible option is to additionally restrict `[[Call]]`, `[[Construct]]` and maybe `[[OwnPropertyKeys]]`. So if there will be a need to test that something doesn't try to call any nested value, we should reintroduce function support. For now I don't know if we *actually* need this. :)